apps/dgst.c

```diff
@@ -407,10 +407,27 @@ int dgst_main(int argc, char **argv)
             BIO_printf(bio_err, "Length can only be specified for XOF\n");
             goto end;
         }
+        /*
+         * Signing using XOF is not supported by any algorithms currently since
+         * each algorithm only calls EVP_DigestFinal_ex() in their sign_final
+         * and verify_final methods.
+         */
         if (sigkey != NULL) {
             BIO_printf(bio_err, "Signing key cannot be specified for XOF\n");
             goto end;
         }
+    } else {
+        /*
+         * Print a warning if the default security strength is less than 128 bits,
+         * but don't fail if the -xoflen was not specified.
+         */
+        if ((EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF)
+                && EVP_MD_get_size(md) < 32)
+            BIO_printf(bio_err,
+                       "Warning: The -xoflen option was not set.\n"
```

Comment:
I'd be tempted to omit the length check entirely.

SHAKE256 is misleading in exactly the same way as SHAKE128.

Reply:
My reasoning is that, in general a security strength of 128 bits is currently acceptable..

Reply:
I am aware of that :) That fixes the SHAKE128 issue .. Anyone using SHA256 will have a similar problem.

Reply:
Coming up with the error message is a bit harder,,,

Reply:
updated